collector/generatorreceiver/internal/topology/metric.go

```diff
@@ -9,6 +9,43 @@ import (
 
 const DefaultPeriod = 60 * time.Minute
 const DefaultOffset = 0 * time.Minute
+const DefaultMetricTickerPeriod = 1 * time.Second
+
+type ShapeInterface interface {
+	GetValue(phase float64) (float64, float64)
+}
+
+type funcShape struct {
+	shape func(phase float64) (float64, float64)
+}
+
+func (fs *funcShape) GetValue(phase float64) (float64, float64) {
+	return fs.shape(phase)
+}
+
+type leakingShape struct {
+	increase   float64
+	average    ShapeInterface
+	kubernetes *Kubernetes
+	lastPod    string
+}
+
+func (ls *leakingShape) GetValue(phase float64) (float64, float64) {
+	if ls.kubernetes == nil {
+		return ls.average.GetValue(phase)
+	}
+
+	if ls.lastPod != ls.kubernetes.PodName {
+		ls.lastPod = ls.kubernetes.PodName
+		ls.increase = 0
+	} else {
+		ls.increase = ls.increase + float64(DefaultMetricTickerPeriod)/float64(ls.kubernetes.Restart.Every)/2
+	}
```

Comment:
q: given the way `increase` is used later it looks like the output of this shape will be a (jittery) sawtooth. Rather than having two outputs for the `GetValue` function could this one just calculate the right `v` directly? The kubernetes struct would need to keep track of its started timestamp, probably. (I ask partly because it seems simpler but mostly because I have some vague plans for composable shapes down the line that depend on the signature being float -> float.)

This isn't urgent, though; unless you feel like it and it seems like a quick change, please don't worry about it. I can tinker with it later :)

Reply:
Def a lot simpler after changing it to be a `float -> float`, thanks for suggesting this 👍